src/main/java/com/google/sps/servlets/PhotosServlet.java

```diff
@@ -0,0 +1,181 @@
+    PreparedQuery results = datastore.prepare(query);
+
+    // Store photos in an array.
+    ArrayList<BlurImage> photos = new ArrayList<>();
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxPhotos))) {
+      long id = entity.getKey().getId();
+      BlobKey blobKey = (BlobKey) entity.getProperty("blobKey");
+      String jsonBlurRectangles = (String) entity.getProperty("jsonBlurRectangles");
+      Date dateCreated = (Date) entity.getProperty("dateCreated");
+
+      photos.add(new BlurImage(id, userId, blobKey, jsonBlurRectangles, dateCreated));
+    }
+
+    // Convert the photos array to JSON.
+    Gson gson = new Gson();
+    String json = gson.toJson(photos);
+
+    // Send the JSON as the response.
+    response.setCharacterEncoding("UTF-8");
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Method to handle the DELETE requests to "/photos" path. Receives "photo-id" parameter. Deletes
+   * a photo from the database.
+   */
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    // If the user is not logged in, send an error message.
+    User user = User.getCurrentUser();
+    if (!user.isLoggedIn()) {
+      response.setContentType("text/html;");
+      response.getWriter().println("You must be logged in to delete photos!");
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+    LoggedUser loggedUser = (LoggedUser) user;
+
+    // Get the input from the request.
+    String idString = request.getParameter("photo-id");
+
+    // Convert the input to a long or show error if the parameter is not a number.
+    long id;
+    try {
+      id = Long.parseLong(idString);
+    } catch (NumberFormatException e) {
+      response.setContentType("text/html;");
+      response.getWriter().println("Parameter photo-id must be a number. Received:" + idString);
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    // Get the photo entity from database.
+    String userId = loggedUser.getId();
+    Query query = new Query("BlurImage");
+    Filter idFilter =
+        new FilterPredicate(
+            Entity.KEY_RESERVED_PROPERTY,
+            FilterOperator.EQUAL,
+            KeyFactory.createKey("BlurImage", id));
+    // We also check the photo's userId to match to the user making the request to avoid deleting
+    // other users' photos.
+    Filter userIdFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+    // Combine the two filters into one.
+    CompositeFilter combinedFilter = CompositeFilterOperator.and(idFilter, userIdFilter);
+    // Apply the filter to the query.
+    query.setFilter(combinedFilter);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    Entity photoEntity = results.asSingleEntity();
```

Comment:
Nit: Could it be a method of the LoggedUser class?

Reply:
Good idea! I moved the logic to LoggedUser class.